### PR TITLE
Expose method to register search strategy routes in query enhancement

### PR DESCRIPTION
**Pull request by amsiglan**
### Description
If a new search strategy is registered, we need a way to setup its routes for the search interceptor to call. This PR exposes a setup method to be able to define the routes for the provided strategy

## Changelog

- fix: Expose method to register search strategy routes in query enhancement

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/query_enhancements/server/routes/index.ts [hunk 1]
```diff
@@ -12,151 +12,130 @@ import {
...
-    {
-      path,
-      validate: {
-        body: schema.object({
-          query: schema.object({
-            query: schema.string(),
-            language: schema.string(),
-            dataset: schema.nullable(schema.object({}, { unknowns: 'allow' })),
-            format: schema.string(),
+export function defineSearchStrategyRouteProvider(logger: Logger, router: IRouter) {
```
Comment (ashwin-pc) [inline src/plugins/query_enhancements/server/routes/index.ts:19]: Can we keep this experimental?

File: src/plugins/query_enhancements/server/routes/index.ts [hunk 2]
```diff
@@ -12,151 +12,130 @@ import {
...
+    );
 
-  router.get(
-    {
-      path: `${path}/{queryId}`,
-      validate: {
-        params: schema.object({
-          queryId: schema.string(),
-        }),
+    router.get(
```
Comment (kavilla) [inline src/plugins/query_enhancements/server/routes/index.ts:82]: headsup git might get confused as i plan on deleting most of these unused methods here: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/8226

File: src/plugins/query_enhancements/server/types.ts [hunk 3]
```diff
@@ -3,11 +3,16 @@
...
-import { PluginSetup } from 'src/plugins/data/server';
+import { ISearchStrategy, PluginSetup } from 'src/plugins/data/server';
 import { DataSourcePluginSetup } from 'src/plugins/data_source/server';
+import { IDataFrameResponse, IOpenSearchDashboardsSearchRequest } from '../../data/common';
 
-// eslint-disable-next-line @typescript-eslint/no-empty-interface
-export interface QueryEnhancementsPluginSetup {}
+export interface QueryEnhancementsPluginSetup {
+  defineSearchStrategyRoute: (
+    searchStrategyId: string,
```
Comment (kavilla) [inline src/plugins/query_enhancements/server/types.ts:12]: nit: could just be `id`

File: src/plugins/query_enhancements/server/types.ts [hunk 4]
```diff
@@ -3,11 +3,16 @@
...
  */
 
-import { PluginSetup } from 'src/plugins/data/server';
+import { ISearchStrategy, PluginSetup } from 'src/plugins/data/server';
 import { DataSourcePluginSetup } from 'src/plugins/data_source/server';
+import { IDataFrameResponse, IOpenSearchDashboardsSearchRequest } from '../../data/common';
 
-// eslint-disable-next-line @typescript-eslint/no-empty-interface
-export interface QueryEnhancementsPluginSetup {}
+export interface QueryEnhancementsPluginSetup {
```
Comment (kavilla) [inline src/plugins/query_enhancements/server/types.ts:10]: thanks for updating the interface and deleting the linter disabler! i think we have a couple of these throughout the project that are left there for no reason

**Comment by kavilla** [comment]
do we need a changelog?

File: src/plugins/query_enhancements/server/routes/index.ts (hunk 2 above)
Comment (amsiglan) [reply src/plugins/query_enhancements/server/routes/index.ts:82]: Do you want to decide which PR goes in first?

File: src/plugins/query_enhancements/server/routes/index.ts (hunk 1 above)
Comment (amsiglan) [reply src/plugins/query_enhancements/server/routes/index.ts:19]: You mean have a comment saying experimental?

**Comment by ashwin-pc** [comment]
We merged #8226. Can you rebase on top of the latest changes in main?

File: src/plugins/query_enhancements/server/routes/index.ts (hunk 2 above)
Comment (amsiglan) [reply src/plugins/query_enhancements/server/routes/index.ts]: resolved the conflicts since 8226 went in

**Review by kavilla — DISMISSED** [review]

**Review by kavilla — APPROVED** [review]

**Review by ashwin-pc — APPROVED** [review]

**Pull request merged**